src/sqlfluff/rules/L046.py

```diff
@@ -54,30 +54,32 @@ def _get_whitespace_ends(s: str) -> Tuple[str, str, str]:
         pos = main.find(inner)
         return main[:pos], inner, main[pos + len(inner) :]
 
-    def _eval(self, context: RuleContext) -> LintResult:
+    def _eval(self, context: RuleContext) -> EvalResultType:
         """Look for non-literal segments."""
         assert context.segment.pos_marker
         if context.segment.is_raw() and not context.segment.pos_marker.is_literal():
+            if not context.memory:
+                memory = set()
+            else:
+                memory = context.memory
+
             # Does it actually look like a tag?
             templated_raw_slices = context.functional.segment.raw_slices.select(
                 rsp.is_slice_type("templated")
             )
+            result = []
```

Comment:
Basically, this fix changes the rule so it loops over **all** the slices of a segment, rather than `return`-ing after the first.